blob/blob_test.go

```diff
@@ -89,10 +98,13 @@ func TestBlob(t *testing.T) {
 	}
 }
 
-func convertBlobs(appBlobs ...types.Blob) ([]*Blob, error) {
+func convertBlobs(appBlobs ...*blob.Blob) ([]*Blob, error) {
 	blobs := make([]*Blob, 0, len(appBlobs))
-	for _, b := range appBlobs {
-		blob, err := NewBlob(b.ShareVersion, append([]byte{b.NamespaceVersion}, b.NamespaceID...), b.Data)
+	for _, appBlob := range appBlobs {
+		if shareVersion := appBlob.GetShareVersion(); shareVersion > math.MaxUint8 {
+			return nil, fmt.Errorf("share version %d is greater than max share version %d", shareVersion, math.MaxUint8)
```

Comment:
Did you trigger this case to make a check for it? Isn't this check part of `NewBlob`

Reply:
it seemed dangerous to blindly cast an `appBlob.GetShareVersion()` to uint8 without testing for overflow  but I can remove this check if you prefer. 

You're right `convertBlobs` is only used in test files.

Reply:
Isn't this check part of `NewBlob`? No because `NewBlob` has a uint8 parameter so the overflow would occur before executing the contents of `NewBlob`.

Reply:
Removed the overflow check in 167496a6020c8525ea59660a7ed3ef373255fa11